src/analyzers/concept/elyses-analytic-enchantments/index.ts

```diff
@@ -0,0 +1,178 @@
+import {
+  AstParser,
+  Input,
+  NoExportError,
+  NoMethodError,
+} from '@exercism/static-analysis'
+import { TSESTree } from '@typescript-eslint/typescript-estree'
+import { CommentType, factory } from '../../../comments/comment'
+import {
+  EXEMPLAR_SOLUTION,
+  FUNCTION_NOT_OPTIMAL,
+  NO_METHOD,
+  NO_NAMED_EXPORT,
+  PREFER_BUILT_IN_METHOD,
+} from '../../../comments/shared'
+import { WritableOutput, ExecutionOptions } from '../../../interface'
+import { IsolatedAnalyzerImpl } from '../../IsolatedAnalyzerImpl'
+import {
+  DOES_STACK_INCLUDE_CARD,
+  DOES_STACK_INCLUDE_ODD_CARD,
+  ElysesAnalyticEnchantmentsSolution,
+  GET_CARD_POSITION,
+  GET_FIRST_EVEN_CARD_POSITION,
+  GET_FIRST_ODD_CARD,
+  IS_EACH_CARD_EVEN,
+} from './ElysesAnalyticEnchantmentsSolution'
+
+type Program = TSESTree.Program
+
+export const MUST_AVOID_IMPERATIVE_LOOP = factory`
+  Avoid use of for loops with this practice exercise.
+
+  According to the instructions, there are build-in methods on the JavaScript
+  Array global object that are geared towards analysis.
+
+  Take a look at the instructions and/or the MDN docs on what methods are
+  available to use.
+`(
+  'javascript.elyses-analytic-enchantments.must_avoid_imperative_loop',
+  CommentType.Essential
+)
+
+export class ElysesAnalyticEnchantmentsAnalyzer extends IsolatedAnalyzerImpl {
+  private solution!: ElysesAnalyticEnchantmentsSolution
+
+  protected async execute(
+    input: Input,
+    output: WritableOutput,
+    options: ExecutionOptions
+  ): Promise<void> {
+    const [parsed] = await AstParser.ANALYZER.parse(input)
+
+    this.solution = this.checkStructure(parsed.program, parsed.source, output)
+    this.solution.readExemplar(options.inputDir)
+
+    if (this.solution.isExemplar) {
+      output.add(EXEMPLAR_SOLUTION())
+      output.finish()
+    }
+
+    if (this.solution.hasforEach || this.solution.hasFor) {
```

Comment:
Perhaps it makes sense to update this to `this.solution.hasImperativeLoop`? That way, if there are new ways/different ways to loop we want to check, it's a single method to update.